src/test/ui/generic-associated-types/issue-86787.rs

```diff
@@ -0,0 +1,40 @@
+#![feature(generic_associated_types)]
```

Comment:
So...no. The suggestion here doesn't actually result in code that compiles, because it causes the where clauses not to match the trait's where clauses. (In fact, in this case, we *can't* write this where clause in the trait, we would have to use `Self`)

Unfortunately, that's a whole separate issue with GATs. But applying the current suggestion should help, in the future, to guide the user towards matching the trait's where clauses.